src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -0,0 +1,546 @@
+            return PauliSum([self, other]).simplify()
+
+        return self + PauliTerm("I0", other)
+
+    def __radd__(self, other: CoefficientTypes) -> "PauliSum":
+        return self + PauliTerm("I0", other)
+
+    def __sub__(
+        self, other: Union[PauliRepresentation, CoefficientTypes]
+    ) -> "PauliSum":
+        return self + -1.0 * other
+
+    def __rsub__(
+        self, other: Union[PauliRepresentation, CoefficientTypes]
+    ) -> "PauliSum":
+        return other + -1.0 * self
+
+    def _multiply_by_operator(self, op: str, index: int) -> "PauliTerm":
+        result_term = PauliTerm("I0", 0)
+        result_coeff = self.coefficient
+        result_ops = self._ops.copy()
+
+        if index not in result_ops:
+            # Case 1: qubit not used yet
+            result_ops[index] = op
+        elif result_ops[index] == op:
+            # Case 2: equal operators cancel
+            del result_ops[index]
+        elif op in ["X", "Y", "Z"]:
+            # Case 3: different pauli ops return third op
+            result_ops[index] = OPERATOR_MAP[ord(self[index]) + ord(op)]
+            result_coeff *= COEFF_MAP[self[index] + op]
+
+        result_term._ops = result_ops
+        result_term.coefficient = result_coeff
+
+        return result_term
+
+    def __mul__(
+        self, other: Union[PauliRepresentation, CoefficientTypes]
+    ) -> PauliRepresentation:
+        """Multiplies this Pauli Term with another PauliTerm, PauliSum, or number
+        according to the Pauli algebra rules.
+        """
+        _validate_type(other)
+
+        if isinstance(other, PauliSum):
+            return (PauliSum([self]) * other).simplify()
+        elif isinstance(other, PauliTerm):
+            result_term = PauliTerm.identity()
+            result_term._ops = self._ops.copy()
+
+            new_coeff = self.coefficient * other.coefficient
+            for op, index in other:
+                if op != "I":
+                    result_term = result_term._multiply_by_operator(op, index)
+
+            return result_term.copy(new_coefficient=result_term.coefficient * new_coeff)
+
+        return self.copy(self.coefficient * complex(other))
+
+    def __rmul__(self, other: CoefficientTypes) -> "PauliTerm":
+        result = self * other
+        assert isinstance(result, PauliTerm)
+        return result
+
+    def __pow__(self, power: int) -> "PauliTerm":
+        """
+        Raises this PauliTerm to power.
+        """
+        if not isinstance(power, int) or power < 0:
+            raise ValueError("The power must be a non-negative integer.")
+
+        return cast(PauliTerm, _efficient_exponentiation(self.copy(), power))
+
+    def __repr__(self) -> str:
+        term_strs = []
+        for index in self._ops.keys():
+            term_strs.append("%s%s" % (self[index], index))
+
+        if len(term_strs) == 0:
+            term_strs.append("I")
+        out = "%s*%s" % (self.coefficient, "*".join(term_strs))
```

Comment:
Are there any reasons we are using the ancient method of string formatting instead of f-strings?? :D

Also, once again, list comprehension would be cleaner.